### PR TITLE
Migrate from Freenode to LiberaChat

### PR DESCRIPTION
**Pull request by vstinner**
Rationale:
https://mail.python.org/archives/list/python-dev@python.org/thread/FQW44V4DZFX4GAGOAMAZAN5KDQXCASDU/

### CODE REVIEW

**Comment by vstinner** [comment]
cc @tiran @matrixise 

@JulienPalard @willingc: are you aware of the #python-doc IRC channel? Are you ok to migrate it as well?

File: communication.rst [hunk 1]
```diff
@@ -50,7 +50,7 @@ to any issue, subscribe to python-bugs-list_.
 
 General Python questions should go to `python-list`_ or `tutor`_
 or similar resources, such as StackOverflow_ or the ``#python`` IRC channel
-on Freenode_.
+on LiberaChat_.
```
Comment (tiran) [inline communication.rst:53]: The official names are "Libera.Chat" or "Libera Chat".

```suggestion
on Libera.Chat_.
```

File: documenting.rst [hunk 2]
```diff
@@ -1736,7 +1736,8 @@ How to get help
 
 Discussions about translations occur on the `doc-sig
 <https://mail.python.org/mailman/listinfo/doc-sig>`_ mailing list,
-and there's a freenode IRC channel, ``#python-doc``.
+and there's a `LiberaChat IRC <https://libera.chat/>`_ channel,
```
Comment (tiran) [inline documenting.rst:1739]:
```suggestion
and there's a `Libera.Chat IRC <https://libera.chat/>`_ channel,
```

File: communication.rst [hunk 3]
```diff
@@ -72,7 +72,7 @@ RSS feed readers.
 .. _python-list: https://mail.python.org/mailman/listinfo/python-list
 .. _tutor: https://mail.python.org/mailman/listinfo/tutor
 .. _StackOverflow: https://stackoverflow.com/
-.. _Freenode: http://freenode.net/
+.. _LiberaChat: https://libera.chat/
```
Comment (encukou) [inline communication.rst:75]:
```suggestion
.. _Libera.Chat: https://libera.chat/
```

File: help.rst [hunk 4]
```diff
@@ -41,7 +41,7 @@ Ask #python-dev
 ---------------
 
 If you are comfortable with IRC you can try asking on ``#python-dev`` (on
-the `freenode`_ network). Typically there are a number of experienced
+the `LiberaChat`_ network). Typically there are a number of experienced
```
Comment (encukou) [inline help.rst:44]:
```suggestion
the `Libera.Chat`_ network). Typically there are a number of experienced
```

File: help.rst [hunk 5]
```diff
@@ -52,7 +52,7 @@ whereas ``#python`` is for questions concerning development *with* Python.
   You may not be able to access the history of this channel, so it cannot
   be used as a "knowledge base" of sorts.
 
-.. _freenode: https://freenode.net/
+.. _LiberaChat: https://libera.chat/
```
Comment (encukou) [inline help.rst:55]:
```suggestion
.. _Libera.Chat: https://libera.chat/
```

**Review by JulienPalard — APPROVED** [review]

**Comment by JulienPalard** [comment]
Thanks @vstinner and everyone proofreading this.

**Pull request merged**